staging/src/k8s.io/apiserver/pkg/authentication/authenticator/interfaces.go

```diff
@@ -26,43 +26,55 @@ import (
 // information about the current user and true if successful, false if not successful,
 // or an error if the token could not be checked.
 type Token interface {
-	AuthenticateToken(token string) (user.Info, bool, error)
+	AuthenticateToken(ctx context.Context, token string) (Response, error)
 }
 
 // Request attempts to extract authentication information from a request and returns
 // information about the current user and true if successful, false if not successful,
 // or an error if the request could not be checked.
 type Request interface {
-	AuthenticateRequest(req *http.Request) (user.Info, bool, error)
+	AuthenticateRequest(ctx context.Context, req *http.Request) (Response, error)
```

Comment:
trying to remember if there was a reason not to use the context associated with the request

Reply:
Since authenticators make RPCs, it seems logical to pass a context to propagate cancellation. It does expand the api surface of authenticators a little bit.

Reply:
`req.Context()` should give you the same context right?

Reply:
Done.